src/middleware/auto-webp.ts

```diff
@@ -0,0 +1,151 @@
+import { proxy } from "../proxy";
+import { Image } from "@fly/v8env/lib/fly/image/index";
+import { get } from "@fly/v8env/lib/fly/cache/response";
+import { set } from "@fly/v8env/lib/fly/cache/response";
+
+/**
+ * Automatically encodes images as webp for supported clients.
+ * This would apply to requests with 'image/webp' in the accept header 
+ * and 'image/jpeg' or 'image/png' in the response type from origin.
+ * 
+ * Example:
+ * 
+ * ```typescript
+ * import { imageService } from "./src/middleware/auto-webp";
+ *
+ * const images = imageService(
+ *   "http://www.example.com/"
+ * )
+ *
+ * fly.http.respondWith(images)
+ * ```
+ */
+
+export interface ImageServiceOptions {
+  rootPath?: string,
+  webp?: boolean
+}
+
+export type FetchFn = (req: RequestInfo, init?: RequestInit) => Promise<Response>
+
+export function imageService(origin: FetchFn | string): FetchFn {
+  return async function imageServiceFetch(req: RequestInfo, init?: RequestInit) {
+    const parser = defaultParser
+    if (typeof origin === "string") {
+      origin = proxy(origin)
+    }
+    if (typeof req === "string") req = new Request(req, init)
+    if (req.method !== "GET" && req.method !== "HEAD") {
```

Comment:
This logic needs to be different for middleware than it is for just an image service. This function rejects everything but get/head requests, which is bad for a middleware context because it might be sitting in front of an app that takes POST requests.

All request should pass through, and there should be logic to decide whether to act on them or not.